pkg/operation/botanist/component/kubescheduler/kube_scheduler.go

```diff
@@ -70,12 +70,37 @@ const (
 	volumeMountPathConfig = "/var/lib/kube-scheduler-config"
 
 	dataKeyComponentConfig = "config.yaml"
-	componentConfigTmpl    = `apiVersion: {{ .apiVersion }}
+
+	componentConfigTmpl = `apiVersion: {{ .apiVersion }}
 kind: KubeSchedulerConfiguration
 clientConnection:
   kubeconfig: ` + gutil.PathGenericKubeconfig + `
 leaderElection:
-  leaderElect: true`
+  leaderElect: true
+{{- if eq .profile "bin-packing" }}
+profiles:
+{{- if or (eq .apiVersion "kubescheduler.config.k8s.io/v1alpha2") (eq .apiVersion "kubescheduler.config.k8s.io/v1beta1") }}
+- schedulerName: bin-packing-scheduler
```

Comment:
Instead of hard-coding the `schedulerName` here, can you introduce a constant and use it? (same below)